mitosheet/src/components/taskpanes/UpdateImports/UpdateImportsUtils.tsx

```diff
@@ -0,0 +1,74 @@
+import { UIState } from "../../../types";
+import { DataframeImportParams } from "../DataframeImport/DataframeImportTaskpane"
+import { CSVImportParams } from "../Import/CSVImport";
+import { ExcelImportParams } from "../Import/XLSXImport";
+import { TaskpaneType } from "../taskpanes";
+import { UpdatedImportObj } from "./UpdateImportsTaskpane"
+
+export interface UpdateImportData {
+    updatedImportObjs: UpdatedImportObj[],
+    importIndex: number
+}
+
+
+export const updateImportedDataWithDataframe = (
+    updateImportedData: UpdateImportData | undefined, 
+    newImportParams: DataframeImportParams,
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>
+): void => {
+    if (updateImportedData === undefined) {
+        return 
+    }
+
+    const newUpdatedImportObjs: UpdatedImportObj[] = JSON.parse(JSON.stringify(updateImportedData.updatedImportObjs))
+    const importIndex = updateImportedData.importIndex
+
+    newUpdatedImportObjs[importIndex] = {
+        ...newUpdatedImportObjs[importIndex],
+        type: 'df', 
+        import_params: newImportParams
+    }
+
+    setUIState(prevUIState => {
+        return {
+            ...prevUIState,
+            currOpenTaskpane: {type: TaskpaneType.UPDATEIMPORTS, updatedImportObjs: newUpdatedImportObjs}
+        }
+    })
+}
+
+export const updateImportedDataWithFile = (
```

Comment:
It's not immediately obvious to me that this couldn't be a single function with the one above -- is there a reason?